mypy/messages.py

```diff
@@ -992,6 +997,27 @@ def pretty_or(args: List[str]) -> str:
     return ", ".join(quoted[:-1]) + ", or " + quoted[-1]
 
 
+def invariance_notes(notes: List[str], arg_type: str, expected_type: str) -> List[str]:
+    """Explain that the type is invariant and give notes for how to solve the issue."""
+    if expected_type.startswith('List') and arg_type.startswith('List'):
```

Comment:
This is still a bit "hacky", I would prefer to check the ``fullname()`` of an actual node, but I will tolerate this, if there is no simple solution.